pkg/certificate/controller.go

```diff
@@ -44,7 +78,8 @@ func (m *Manager) add(mgr manager.Manager, r reconcile.Reconciler) error {
 			return isWebhookConfigOrAnnotatedResource(createEvent.Meta)
 		},
 		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
-			return isWebhookConfigOrAnnotatedResource(deleteEvent.Meta)
+			_, isASecret := deleteEvent.Object.(*corev1.Secret)
```

Comment:
The sematics would be clearer if we have a separate branch for secret, so something like:

```
if isASecret {
    return special logic
}
return false
```

That way it is clear that for non-secrets it is always false without reading into the logic expression

Reply:
And add at least a comment explaining that we do not want to recreate Secrets on deletion of the whole app. We want to recreate it only when they were deleted alone, by mistake or purpose to recreate a new cert.